### ISSUE TITLE
Unity Internal Error after update pushwoosh plugin

### ISSUE THREAD

**Issue by Tavernari** [Pushwoosh/pushwoosh-unity#8]
Hi,

I had a internal error after update, just update and did this stacktrace error
other information is my Unity is 4.7.2f1

I tried fix but nothing happing.

`Internal compiler error. See the console log for more information. output was:
Unhandled Exception: System.Reflection.ReflectionTypeLoadException: The classes in the module cannot be loaded.
  at (wrapper managed-to-native) System.Reflection.Assembly:GetTypes (bool)
  at System.Reflection.Assembly.GetTypes () [0x00000] in <filename unknown>:0 
  at Mono.CSharp.RootNamespace.ComputeNamespaces (System.Reflection.Assembly assembly, System.Type extensionType) [0x00000] in <filename unknown>:0 
  at Mono.CSharp.RootNamespace.ComputeNamespace (Mono.CSharp.CompilerContext ctx, System.Type extensionType) [0x00000] in <filename unknown>:0 
  at Mono.CSharp.GlobalRootNamespace.ComputeNamespaces (Mono.CSharp.CompilerContext ctx) [0x00000] in <filename unknown>:0 
  at Mono.CSharp.Driver.LoadReferences () [0x00000] in <filename unknown>:0 
  at Mono.CSharp.Driver.Compile () [0x00000] in <filename unknown>:0 
  at Mono.CSharp.Driver.Main (System.String[] args) [0x00000] in <filename unknown>:0 
`

and that too

`Internal compiler error. See the console log for more information. output was:
Unhandled Exception: System.IO.FileNotFoundException: Could not load file or assembly 'System.Runtime, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies.
File name: 'System.Runtime, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
  at (wrapper managed-to-native) System.Reflection.Assembly:InternalGetType (System.Reflection.Module,string,bool,bool)
  at System.Reflection.Assembly.GetType (System.String name, Boolean throwOnError, Boolean ignoreCase) [0x00000] in <filename unknown>:0 
  at System.Reflection.Assembly.GetType (System.String name) [0x00000] in <filename unknown>:0 
  at Mono.CSharp.RootNamespace.GetTypeInAssembly (System.Reflection.Assembly assembly, System.String name) [0x00000] in <filename unknown>:0 
  at Mono.CSharp.RootNamespace.LookupTypeReflection (Mono.CSharp.CompilerContext ctx, System.String name, Location loc, Boolean must_be_unique) [0x00000] in <filename unknown>:0 
  at Mono.CSharp.GlobalRootNamespace.LookupTypeReflection (Mono.CSharp.CompilerContext ctx, System.String name, Location loc, Boolean must_be_unique) [0x00000] in <filename unknown>:0 
  at Mono.CSharp.Namespace.LookupType (Mono.CSharp.CompilerContext ctx, System.String name, Location loc) [0x00000] in <filename unknown>:0 
  at Mono.CSharp.Namespace.Lookup (Mono.CSharp.CompilerContext ctx, System.String name, Location loc) [0x00000] in <filename unknown>:0 
  at Mono.CSharp.NamespaceEntry.Lookup (System.String name, Location loc, Boolean ignore_cs0104) [0x00000] in <filename unknown>:0 
  at Mono.CSharp.NamespaceEntry.LookupNamespaceOrType (System.String name, Location loc, Boolean ignore_cs0104) [0x00000] in <filename unknown>:0 
  at Mono.CSharp.DeclSpace.LookupNamespaceOrType (System.String name, Location loc, Boolean ignore_cs0104) [0x00000] in <filename unknown>:0 
  at Mono.CSharp.TypeContainer+BaseContext.LookupNamespaceOrType (System.String name, Location loc, Boolean ignore_cs0104) [0x00000] in <filename unknown>:0 
  at Mono.CSharp.SimpleName.ResolveAsTypeStep (IMemberContext ec, Boolean silent) [0x00000] in <filename unknown>:0 
  at Mono.CSharp.Expression.ResolveAsBaseTerminal (IMemberContext ec, Boolean silent) [0x00000] in <filename unknown>:0 
  at Mono.CSharp.TypeContainer.ResolveBaseTypes (Mono.CSharp.TypeExpr& base_class) [0x00000] in <filename unknown>:0 
  at Mono.CSharp.Class.ResolveBaseTypes (Mono.CSharp.TypeExpr& base_class) [0x00000] in <filename unknown>:0 
  at Mono.CSharp.TypeContainer.DefineBaseTypes () [0x00000] in <filename unknown>:0 
  at Mono.CSharp.TypeContainer.DefineType () [0x00000] in <filename unknown>:0 
  at Mono.CSharp.RootContext.ResolveTree () [0x00000] in <filename unknown>:0 
  at Mono.CSharp.Driver.Compile () [0x00000] in <filename unknown>:0 
  at Mono.CSharp.Driver.Main (System.String[] args) [0x00000] in <filename unknown>:0`

**Comment by DimanAM** [comment]
Does the issue persist if you clean your project? remove Pushwoosh plugin? reinstall Unity/Mono/.NET?
What are the steps to reproduce your problem? What platform are you building?

**Comment by Tavernari** [comment]
Hi @DimanAM 

I did a binary search here and I found last commit without this issue is `f9cb40137f33c3449474db949c7835e3ae71a9cd`

I am using Unity 4.7 with Mono

MonoDevelop Informations

MonoDevelop-Unity
Version 4.0.1
Runtime:
    Mono 2.10.12 (mono-2-10/c9b270d Thu Mar  7 21:38:12 EST 2013)
    GTK 2.24.16
    GTK# (2.12.0.0)
    Package version: 210120000

Build Information
Git revision: Release ID:
0557c5690da014ceb6679b23f2aee1b51df0aa5a
Build date: 2015-06-16 09:23:57+0000

Operating System
Mac OS X 10.11.2

**Comment by Tavernari** [comment]
When I remove the pushwoosh plugin the Unity work without errors, but when I import the plugin crash with this error

**Comment by DimanAM** [comment]
Should be fixed now. Unity 4 does not like the existence of WSA platform in Plugins directory for some reason. Please use Unity4-Pushwoosh-${VERSION_NUMBER}.unitypackage on Unity4.

**Comment by Orfen-0** [comment]
I've had issues with the aar format of the libraries. I know that since 4.2 they are supposed to be compatible, but i had compilation issues that were solved by extracting the classes.jar, renaming them and using those. The next issue i'm facing is with the version.xml file needed for  
`<meta-data android:name="com.google.android.gms.version" android:value="@integer/google_play_services_version" />
entry` in the manifest file. I replaced the value with the static numerical version, but it fails the validation on the control panel. I'm not sure if there are real issues except from the validation failing.
